internal/session/repository_connection.go

```diff
@@ -442,14 +436,22 @@ func (r *ConnectionRepository) closeOrphanedConnections(ctx context.Context, wor
 	return orphanedConns, nil
 }
 
-func fetchConnectionStates(ctx context.Context, r db.Reader, connectionId string, opt ...db.Option) ([]*ConnectionState, error) {
-	const op = "session.fetchConnectionStates"
-	var states []*ConnectionState
-	if err := r.SearchWhere(ctx, &states, "connection_id = ?", []any{connectionId}, opt...); err != nil {
-		return nil, errors.Wrap(ctx, err, op)
+func fetchConnectionStatus(ctx context.Context, r db.Reader, connectionId string, opt ...db.Option) (ConnectionStatus, error) {
+	const op = "session.fetchConnectionStatus"
+	rows, err := r.Query(ctx, fetchConnectionStatusCte, []interface{}{connectionId}, opt...)
```

Comment:
It might be a little cleaner to make a view for retrieving a connection with the status. For example a view like:

```sql
create view session_connection_with_status_view as
     select public_id,
            session_id,
            client_tcp_address,
            client_tcp_port,
            endpoint_tcp_address,
            endpoint_tcp_port,
            bytes_up,
            bytes_down,
            closed_reason,
            version,
            create_time,
            update_time,
            user_client_ip,
            worker_id,
            case
               when connected_time_range is null        then 'authorized'
               when upper(connected_time_range) > now() then 'connected'
               else                                          'closed'
            end as status
       from session_connection;
```

Then we can eliminate this `fetchConnectionStates` function and `fetchConnectionStausCte` (which is also a misnomer since it is not a CTE). Instead we can just use `LookupById` and the status would appear just like any other column in the sql result.